analyzers/src/SonarAnalyzer.CSharp/Rules/ParameterValidationInMethodWalker.cs

```diff
@@ -61,16 +61,22 @@ public override void VisitThrowStatement(ThrowStatementSyntax node)
             {
                 argumentExceptionLocations.Add(node.Expression.GetLocation());
             }
+
             // there is no need to visit children
         }
 
         public override void VisitInvocationExpression(InvocationExpressionSyntax node)
         {
             if (node.IsMemberAccessOnKnownType("ThrowIfNull", KnownType.System_ArgumentNullException, semanticModel))
             {
+                // "ThrowIfNull" returns void so it cannot be an argument. We can stop.
                 argumentExceptionLocations.Add(node.GetLocation());
             }
-            // "ThrowIfNull" returns void so it cannot be an argument. We can stop.
+            else
```

Comment:
Nitpick and completely optional, as you will "work" on `ParameterValidationInYieldShouldBeWrapped` by adding test cases I would also clean up the implementation of the rule while you have the context.